### PR TITLE
Overhaul to Ice River mission

### PR DESCRIPTION
**Pull request by OcelotWalrus**

### CODE REVIEW

**Comment by Dzierzan** [comment]
So... patrolling wasn't possible in the end? I am not sure if removing almost all enemy actors was a good idea.

File: mods/hv/maps/ice-river/map.yaml [hunk 1]
```diff
@@ -211,6 +153,37 @@ Actors:
...
+	Actor36: electricpod
+		Owner: Yuruki Industries
+		SubCell: 1
+		Facing: 384
+		Location: 50,52
+	Actor37: saucer
+		Owner: Yuruki Industries
+		Facing: 384
+		Location: 42,55
+	Actor38: flyingmonster
```
Comment (Dzierzan) [inline mods/hv/maps/ice-river/map.yaml:183]: Nah, don't add force adding critter.

**Review by Dzierzan — CHANGES_REQUESTED** [review]

File: mods/hv/maps/ice-river/map.yaml (hunk 1 above)
Comment (OcelotWalrus) [reply mods/hv/maps/ice-river/map.yaml:183]: ?

**Comment by OcelotWalrus** [comment]
> So... patrolling wasn't possible in the end? I am not sure if removing almost all enemy actors was a good idea.

It's possible, but anyway the enemy will periodically produce units. But there can be patrolling units if you want.

File: mods/hv/maps/ice-river/map.yaml (hunk 1 above)
Comment (Dzierzan) [reply mods/hv/maps/ice-river/map.yaml:183]: Remove that critter please.

**Comment by Dzierzan** [comment]
> > So... patrolling wasn't possible in the end? I am not sure if removing almost all enemy actors was a good idea.
> 
> It's possible, but anyway the enemy will periodically produce units. But there can be patrolling units if you want.

I did like some defenders in the first iteration of this mission. I know AI will use them as soon as it activates. You could technically add 2nd enemy player with the same color with brain dead AI (just enemy with no AI logic applied, kinda like Creep faction), so 1st enemy will never use them. A bit hacky, but it could work.

**Comment by OcelotWalrus** [comment]
>  I know AI will use them as soon as it activates

The AI won't, as they're ordered to patrol through the script. I added some patrolling units on this force push.

File: mods/hv/maps/ice-river/map.yaml (hunk 1 above)
Comment (OcelotWalrus) [reply mods/hv/maps/ice-river/map.yaml]: Done

File: mods/hv/maps/ice-river/map.yaml [hunk 2]
```diff
@@ -211,6 +153,42 @@ Actors:
...
+		Owner: Yuruki Industries
+		SubCell: 1
+		Location: 44,49
+		Facing: 493
+	Actor36: electricpod
+		Owner: Yuruki Industries
+		SubCell: 1
+		Facing: 384
+		Location: 50,52
+	Actor37: saucer
```
Comment (Dzierzan) [inline mods/hv/maps/ice-river/map.yaml:176]: Please, remove the saucer.

**Review by Dzierzan — CHANGES_REQUESTED** [review]

**Comment by Dzierzan** [comment]
image
Two types of MG pods are available.

**Comment by Dzierzan** [comment]
Remove AA tank from Factory and Trade Platform. The enemy faction has only one balloon and you can destroy that with electric pods anyway.

I think Repair Tank could be actually readded.

**Comment by OcelotWalrus** [comment]
Repair Tank hasn't been disabled.

**Comment by Dzierzan** [comment]
Right, I forgot it's in Utility tab.

**Comment by OcelotWalrus** [comment]
Is it good for you @Dzierzan ?

**Comment by Dzierzan** [comment]
AI can build more miners than needed:
image

I noticed patrolling enemy units do not attack back.

**Comment by OcelotWalrus** [comment]
> AI can build more miners than needed:

That's because the periodic productions the AI is ordered to do by the script contain a miner. I don't really see a problem in that in my opinion.

> I noticed patrolling enemy units do not attack back.

That's weird because they do for me. Are you sure you updated your branch?

**Comment by Dzierzan** [comment]
Then the script needs to take in account having already existing miner unit on the battlefield, can you try to update it? 

Yes, I am sure my branch was up to date.

**Comment by OcelotWalrus** [comment]
> Then the script needs to take in account having already existing miner unit on the battlefield, can you try to update it?

I'll make so if a miner already exists (because it hasn't been sent as there's no more avaible resource spots), the enemy won't produce it.

> Yes, I am sure my branch was up to date.

They do attack back for me.

https://github.com/user-attachments/assets/490b3dd7-8019-42a2-a21c-9e79e28462d3

**Comment by OcelotWalrus** [comment]
I understood why they didn't attack you back. When patrolling, units wait a defined amount of time at the waypoint they're in. During that time they can't attack (idk why). So I set that amount to 0 to fix that.

**Pull request merged**